### DISCUSSION TITLE
Bottom labels not rotating when needed

### DISCUSSION THREAD

**Discussion by X-lem** [TanStack/react-charts#206] [Q&A] [answered]
I have a bar graph that users can switch between viewing the data from weekly to daily. When I switch to the daily the bottom labels don't rotate so they're readable as shown in the sandbox examples. The examples don't show any special code I need to add in order to achieve this so I'm unsure why I'm not seeing similar behavior.

**Weekly**
![image](https://user-images.githubusercontent.com/6555273/122956392-cebaad00-d335-11eb-816a-56a14a481c6d.png)

**Daily**
![image](https://user-images.githubusercontent.com/6555273/122956501-e8f48b00-d335-11eb-8e2d-7159fd1272ca.png)

**My code (without import statements)**

_Call_
Note: `this.state.totalServerData` is being updated when switching from weekly to daily view.
`<BarGraph cssClass={"verticalBarGraph"} data={this.state.totalServerData} primaryPosition={"bottom"} secondaryPosition={"left"} />`

_Function_
```
export default function BarGraph(props) {

    const series = React.useMemo(
        () => ({
          type: "bar"
        }),
        []
    );

    const axes = React.useMemo(
        () => [
          { primary: true, type: "ordinal", position: props.primaryPosition },
          { position: props.secondaryPosition, type: "linear", stacked: true }
        ],
        []
    );

    const data = React.useMemo(
        () => props.data,
        [props.data]
    )

    const tooltip = React.useMemo(
        () => ({
          anchor: "gridRight"
        }),
        []
    )

    let className = 'barGraph ';
    if (props.cssClass) {
        className += props.cssClass;
    }

    return (
        <div className={className}>
            <Chart data={data} axes={axes} series={series} tooltip={tooltip} />
        </div>
    )
}
```

**Comment by tannerlinsley** [comment]
I'll need to see a codesandbox example to accurately debug this.

**Reply by X-lem** [reply]
Created a codesandbox example for you. The only main difference is that I'm calling an API (my DB) in order to retrieve the data. Otherwise it's pretty much the same. Let me know if you can't see it.

[https://codesandbox.io/s/cool-firefly-otyih](https://codesandbox.io/s/cool-firefly-otyih)

**Comment by tannerlinsley** [comment, marked as answer]
Due to a massive overhaul and rewrite of React Charts to TypeScript and a massively upgraded public API, this discussion is being marked as answered for triage. The latest version can be tested via the `react-charts@beta` tag. Documentation is in the process of being upgraded, but thankfully, everything is fully typed now, so you can explore the API that way for now.

If, *after testing or playing with the v3 beta*, you believe this discussion is still relevant or unanswered, feel free to mark it as such and continue the discussion.
